src/connection/index.ts

```diff
@@ -25,10 +25,21 @@ export interface Connection {
   type: ConnectionType
 }
 
+let metamaskErrorHandler: (error: Error) => void | undefined
+
+export function setErrorHandler(errorHandler: (error: Error) => void) {
```

Comment:
This should be named more descriptively: `setMetaMaskErrorHandler`. Also, note that `MetaMask` has two capital `M`s for naming (eg on L28).

Reply:
sg done